Tests/ApolloTests/StarWars/API.swift

```diff
@@ -336,65 +406,82 @@ public final class HeroDetailsQuery: GraphQLQuery {
   public struct Data: GraphQLMappable {
     public let hero: Hero?
 
-    public init(reader: GraphQLResultReader) throws {
-      hero = try reader.optionalValue(for: Field(responseName: "hero", arguments: ["episode": reader.variables["episode"]]))
+    public init(values: [Any?]) {
+      hero = values[0] as! Hero?
     }
 
     public struct Hero: GraphQLMappable {
       public let __typename: String
       public let name: String
-
       public let asHuman: AsHuman?
       public let asDroid: AsDroid?
 
-      public init(reader: GraphQLResultReader) throws {
-        __typename = try reader.value(for: Field(responseName: "__typename"))
-        name = try reader.value(for: Field(responseName: "name"))
-
-        asHuman = try AsHuman(reader: reader, ifTypeMatches: __typename)
-        asDroid = try AsDroid(reader: reader, ifTypeMatches: __typename)
+      public init(values: [Any?]) {
+        __typename = values[0] as! String
+        name = values[1] as! String
+        asHuman = values[2] as! AsHuman?
```

Comment:
Does the user access data in fragments like: `hero.asHuman.height` (with any appropriate type checks of course). I’d love to know why this was done and the general rules around how it works. What are the other approaches? Does Swift have a really good enum/data type like Rust or Haskell that works well with exhaustive pattern matching?